### ISSUE TITLE
FigureCanvasPdf object has no attribute get_renderer

### ISSUE THREAD

**Issue by marcelm** [mwaskom/seaborn#1262]
Hi, commit 7b445822bc4c1edf22c8e8e11f67e632ffb90f86 introduced a regression when plotting a clustermap to a PDF.

The following code (based on the example code) works when using seaborn 0.8.0:
```python
import matplotlib
matplotlib.use('pdf')
import seaborn as sns
sns.set(color_codes=True)
iris = sns.load_dataset("iris")
species = iris.pop("species")
g = sns.clustermap(iris)
g.savefig("out.pdf")
```
On master, the following error is raised:
```
Traceback (most recent call last):
  File "plot.py", line 7, in <module>
    g = sns.clustermap(iris)
  File ".../seaborn/matrix.py", line 1299, in clustermap
    **kwargs)
  File ".../seaborn/matrix.py", line 1129, in plot
    row_linkage=row_linkage, col_linkage=col_linkage)
  File ".../seaborn/matrix.py", line 1024, in plot_dendrograms
    ax=self.ax_row_dendrogram, rotate=True, linkage=row_linkage)
  File ".../seaborn/matrix.py", line 749, in dendrogram
    return plotter.plot(ax=ax)
  File ".../seaborn/matrix.py", line 700, in plot
    ax.figure.draw(ax.figure.canvas.get_renderer())
AttributeError: 'FigureCanvasPdf' object has no attribute 'get_renderer'
```
I need to choose the pdf backend at the beginning to ensure that I don’t get error messages when I run the script non-interactively (no `DISPLAY` variable).

**Comment by mwaskom** [comment]
> I need to choose the pdf backend at the beginning to ensure that I don’t get error messages when I run the script non-interactively (no DISPLAY variable).

Try the `Agg` backend. Also you'll want to rasterize the heatmap if you're plotting to pdf.

**Comment by marcelm** [comment]
Thanks, it works with `use('agg')`. Thanks also for recommending `rasterized=True`, I had forgotten about that!

(If I understand correctly, matplotlib switches to the pdf backend anyway during `g.savefig("out.pdf")`, so it’s a bit weird this works.)

**Comment by blaiseli** [comment]
I'm having the same issue after updating seaborn, and I'm not sure where in my code I should switch backend to "Agg" (or should it be "agg" ?) in my complex setting: The part where I use `clustermap` belongs to a large snakemake workflow where other figures are generated using matplotlib (some of them using tex for text rendering, if that matters), and so far everything worked fine with `matplotlib.use("PDF")` at the beginning of my code, so I'm afraid of breaking stuff somewhere else if I change the backend globally. I tried to put `matplotlib.use("agg")` just before calling `clustermap`, but I get the following warning:
```
This call to matplotlib.use() has no effect because the backend has already
been chosen; matplotlib.use() must be called *before* pylab, matplotlib.pyplot,
or matplotlib.backends is imported for the first time.
```

And then, the `AttributeError`.

I tried to change globally, and this solves the `AttributeError`, but, as I feared, other problems arise. It seems that "Agg" interferes with the usage of `matplotlib.backends.backend_pdf.PdfPages`
